src/main/scala/nl.knaw.dans.easy.deposit/StateManager.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit
+
+import java.util.UUID
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.Errors.{ IllegalStateTransitionException, InvalidPropertyException, PropertyNotFoundException }
+import nl.knaw.dans.easy.deposit.docs.StateInfo
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State.State
+import org.apache.commons.configuration.PropertiesConfiguration
+
+import scala.util.{ Failure, Success, Try }
+
+case class StateManager(depositDir: File, submitBase: File) {
+  private val stateDescriptionKey = "state.description"
+  private val stateLabelKey = "state.label"
+  private val bagIdKey = "bag-store.bag-id"
+  val draftProps = new PropertiesConfiguration(
+    (depositDir / "deposit.properties").toJava
+  )
+  private lazy val submittedProps = new PropertiesConfiguration(
+    (submitBase / getProp(bagIdKey, draftProps) / "deposit.properties").toJava
+  )
+
+  def getStateInfo: Try[StateInfo] = Try {
+    val draftState = getStateLabel(draftProps)
+    draftState match {
+      case State.submitted | State.inProgress =>
+        val newState: StateInfo = getProp(stateLabelKey, submittedProps) match {
+          case "SUBMITTED" => StateInfo(State.submitted, getStateDescription(draftProps))
+          case "REJECTED" => StateInfo(State.rejected, getStateDescription(submittedProps))
+          case "FEDORA_ARCHIVED" => StateInfo(State.archived, "The dataset is published")
+          case "IN_REVIEW" => StateInfo(State.inProgress, "The dataset is visible for you under your datasets in https://easy.dans.knaw.nl/ui")
```

Comment:
what happens if we put an `<a>` on this string? does it render as a clickable url in deposit-ui or does it get escaped?

Reply:
The html-tag is displayed as plain text

Reply:
There is a way to render this as html, but I'm not yet sure if we should use this method...

https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml

Reply:
It's dangerous when you put user input in innerHTML. In this case you are not echoing user input. Nevertheless extending the returned json seems a nicer approach IMHO.